### DISCUSSION TITLE
ValueError : Bounding Box Debugger/Images is invalid. Please remove invalid filename characters

### DISCUSSION THREAD

**Discussion by An0rangetree** [ultralytics/yolov5#5308] [Q&A]
![Snipaste_2021-10-23_18-07-28](https://user-images.githubusercontent.com/87368283/138551994-f065e078-fe30-4f9a-b446-024df4d36148.png)

ValueError : Bounding Box Debugger/Images is invalid. Please remove invalid filename characters

Hi, I am new to yolov5.

When I was doing model training, this error interrupted the process. Where can I find the "invalid filename characters" that caused this problem? How can I fix it?

I have successfully run the yolov5s model with similar configuration on other projects. Recently, I want to apply it to other projects. I don't know what caused the mistake. Maybe it's because of yolov5 update? Or did I make a new bug in the configuration file? Or is there a problem with the picture set used in the new project? This problem should have nothing to do with the running environment.

Thank you in advance.

**Comment by glenn-jocher** [comment]
@An0rangetree can you provide code and dataset to help us reproduce this issue? Has this happened to you before on the same dataset?

@AyushExel seems like a new wandb bug here related to image filename characters. Can you take a look at this please?

EDIT: @AyushExel adding a TODO to this so we don't lose track

**Reply by AyushExel** [reply]
@glenn-jocher this seems P0. Looking into it now

**Comment by AyushExel** [comment]
@glenn-jocher @An0rangetree I couldn't reproduce this on colab. Could you tell me what env you're using?

**Reply by glenn-jocher** [reply]
@AyushExel I think you'll need access to @An0rangetree dataset that contains the images with the 'invalid' characters to reproduce.

@An0rangetree can you share your dataset to help us reproduce this?

**Comment by JYQmurphy** [comment]
i aslo met this condition，i used 'coco128' to train my model

**Reply by glenn-jocher** [reply]
@JYQmurphy thanks for the feedback! Can you confirm that you can reproduce this with the latest YOLOv5 code on COCO128?

@AyushExel was our recent PR supposed to fix this or is that a seperate issue?

**Reply by AyushExel** [reply]
It should be fixed by the PR. Are you still seeing this after pulling the latest changes ?

**Reply by JYQmurphy** [reply]
i haven't try yet, but in a website in China i found a solution, it is said i should recreate an environment (use anaconda), so im now tring this way, i will inform you guys wahtever it comes out. thx

**Reply by JYQmurphy** [reply]
it is said that the database is cool, just because the wandb wasn‘t installed in right way

**Reply by AyushExel** [reply]
Let us know if the problem still exists. Thanks!

**Reply by JYQmurphy** [reply]
i used the latest YOLOv5 and it seems like the problem is fixed，but actually i don't know whether it is because i create a new environment. Btw i changed ‘—workers’ to 0. Thanks for help, have a nice day

**Reply by AyushExel** [reply]
thanks for the feedback :+1:

**Comment by endlessBUG** [comment]
impact version: wandb==0.5.8
found working after commenting/removing below code in python function wandb.sdk.data_types.Media.bind_to_run

if SYS_PLATFORM == "Windows" and not util.check_windows_valid_filename(key):
    raise ValueError(
        f"Media {key} is invalid. Please remove invalid filename characters"
    )

**Comment by Olalaye** [comment]
```

     Epoch   gpu_mem       box       obj       cls    labels  img_size
    29/299     18.8G   0.02747    0.0497   0.00741       220      1280: 100%|████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:03<00:00,  2.49it/s]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|████████████████████████████████████████████████████████████████████████| 4/4 [00:06<00:00,  1.55s/it] 
                 all        128        929      0.861      0.837      0.911       0.63
Traceback (most recent call last):
  File "train.py", line 621, in <module>
    main(opt)
  File "train.py", line 518, in main
    train(opt.hyp, opt, device, callbacks)
  File "train.py", line 370, in train
    callbacks.run('on_fit_epoch_end', log_vals, epoch, best_fitness, fi)
  File "E:\data\yolov5\utils\callbacks.py", line 76, in run
    logger['callback'](*args, **kwargs)
  File "E:\data\yolov5\utils\loggers\__init__.py", line 119, in on_fit_epoch_end
    self.wandb.end_epoch(best_result=best_fitness == fi)
  File "E:\data\yolov5\utils\loggers\wandb\wandb_utils.py", line 493, in end_epoch
    wandb.log(self.log_dict)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_run.py", line 1229, in log
    self.history._row_add(data)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_history.py", line 44, in _row_add
    self._flush()
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_history.py", line 59, in _flush
    self._callback(row=self._data, step=self._step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_run.py", line 972, in _history_callback
    self._backend.interface.publish_history(
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\interface\interface.py", line 491, in publish_history
    data = data_types.history_dict_to_json(run, data, step=step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 2325, in history_dict_to_json
    payload[key] = val_to_json(run, key, val, namespace=step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 2366, in val_to_json
    item.bind_to_run(run, key, namespace, id_=i)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 1984, in bind_to_run
    super(Image, self).bind_to_run(run, key, step, id_)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 494, in bind_to_run
    raise ValueError(
ValueError: Media Bounding Box Debugger/Images is invalid. Please remove invalid filename characters
wandb: ERROR Problem finishing run
Traceback (most recent call last):
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_run.py", line 1671, in _atexit_cleanup
    self._on_finish()
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_run.py", line 1815, in _on_finish
    self.history._flush()
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_history.py", line 59, in _flush
    self._callback(row=self._data, step=self._step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\wandb_run.py", line 972, in _history_callback
    self._backend.interface.publish_history(
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\interface\interface.py", line 491, in publish_history
    data = data_types.history_dict_to_json(run, data, step=step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 2325, in history_dict_to_json
    payload[key] = val_to_json(run, key, val, namespace=step)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 2366, in val_to_json
    item.bind_to_run(run, key, namespace, id_=i)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 1984, in bind_to_run
    super(Image, self).bind_to_run(run, key, step, id_)
  File "d:\Anaconda3\envs\yolo2\lib\site-packages\wandb\sdk\data_types.py", line 494, in bind_to_run
    raise ValueError(
ValueError: Media Bounding Box Debugger/Images is invalid. Please remove invalid filename characters

(yolo2) E:\data\yolov5>pip show wandb
Name: wandb
Version: 0.12.6
Summary: A CLI and library for interacting with the Weights and Biases API.
Home-page: https://github.com/wandb/client
Author: Weights & Biases
Author-email: support@wandb.com
License: MIT license
Location: d:\anaconda3\envs\yolo2\lib\site-packages
Requires: Click, configparser, docker-pycreds, GitPython, pathtools, promise, protobuf, psutil, python-dateutil, PyYAML, requests, sentry-sdk, shortuuid, six, subprocess32, yaspin
Required-by:
```
**OS:** Windows 10
**wandb:** Version: 0.12.6
**dataset:** 'coco128'
**Code**：
```
(yolo2) E:\data\yolov5>git show
commit 956be8e642b5c10af4a1533e09084ca32ff4f21f (HEAD, tag: v6.0)
```
This error was encountered several times when the EPOCH was 29

**Reply by Olalaye** [reply]
@glenn-jocher What was done when the EPOCH was equal to 29( maybe 30, because it started at 0 )

**Reply by glenn-jocher** [reply]
@Olalaye we don't do anything special on epoch 29, but W&B might. This error appears to originate in the W&B code we have. @AyushExel multiple (4) users are reporting this same bug now. The relevant traceback section is:
```
  File "E:\data\yolov5\utils\loggers\__init__.py", line 119, in on_fit_epoch_end
    self.wandb.end_epoch(best_result=best_fitness == fi)
  File "E:\data\yolov5\utils\loggers\wandb\wandb_utils.py", line 493, in end_epoch
    wandb.log(self.log_dict)
```

I have the same question as @Olalaye, how could we do 29 good epochs and this occurs on the 30th? Is there W&B logic that starts on the 30th epoch?

**Reply by glenn-jocher** [reply]
@Olalaye could you pass us the training command you used (i.e. any non-default settings)? Did you modify the repository code? Is this reproducible (i.e. does it occur again in epoch 29 if you retrain?)

**Reply by glenn-jocher** [reply]
@AyushExel could this be related to boundix box debugger images that get uploaded in 10 sets spread out over --epochs?

**Reply by AyushExel** [reply]
@glenn-jocher this was fixed earlier.. doing a git pull should resolve it.

This happens because the name cannot contain spaces in windows. 
Why does this happen only at 30th epoch?
Because the bbox_interval is set to `epochs//10` by default to make sure we only predictions 10 times. User can change it if they want. Thus by default the 300//10 results to 30 and the first batch is logged at 30th epoch

**Reply by glenn-jocher** [reply]
@AyushExel oh got it, thanks! I'll post a fix message then for future users arriving here and remove the TODO.

**Reply by glenn-jocher** [reply]
@AyushExel do you have a PR I can point to in my fix message?

**Reply by AyushExel** [reply]
@glenn-jocher   This one https://github.com/ultralytics/yolov5/pull/5317

**Comment by glenn-jocher** [comment]
# Issue Fixed

```
ValueError : Bounding Box Debugger/Images is invalid. Please remove invalid filename characters
```

This error message has now be fixed ✅ in PR #5317 by @AyushExel 

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Reply by pengpengdesu** [reply]
thank you very much！！！！
